### PR TITLE
add extra user-agent option

### PR DESCRIPTION
**Pull request by anntisa**
I'm not quite sure how to test what sort of exceptions it comes across if the 1st attempt doesn't work. so I had it just catch all exceptions and then do the second attempt in the nested try catch. 
I know pretty shitty...

If you have any better ideas, or on how to test it, please tell. 🙏 

THis is the email

```
During testing we had this exception: https://gist.github.com/anonymous/a0ddde564379fe75437b0df7ecc9576a . Thanks to AndroidStudio's feature to look inside jar file, I found that "com.survata.utils.Utils.getUserAgent(Utils.java:100)" loads app's name via com.survata.R.string class. Could you change this:

String currentUserAgent = context.getString(string.app_name) + "/" + packageName;

to:

Resources res = context.getResources();
int resId = res.getIdentifier("app_name", "string", context.getPackageName());
String currentUserAgent = res.getString(resId) + "/" + packageName;

?

In case when you can't receive package name you can, use some static text for userAgent.
```

### CODE REVIEW

**Comment by rogueleaderr** [comment]
Isn't Java supposed to tell you all the exceptions that could occur and refuse to compile unless they're handled?

Overall I don't know how to evaluate whether this is reasonable but assuming you've tested it and it doesn't crash it seems okay.

Alternatively, maybe you could make resourceName a default parameter (if Java has default parameters...) so that the user has the option of passing in a static text if they need to. (I.e. make it their problem instead of ours).

File: survata-sdk/src/main/java/com/survata/utils/Utils.java [hunk 1]
```diff
@@ -90,14 +91,27 @@ public static String parseParamMap(Map<String, String> params) {
...
+            resourceName = context.getString(R.string.app_name);
+
         } catch (Exception e) {
-            Logger.e(TAG, "get user agent failed", e);
+            Logger.e(TAG, "1st attempt, user agent failed, trying again...", e);
+            try {
+                Resources res = context.getResources();
+                int resId = res.getIdentifier("app_name", "string", context.getPackageName());
+                resourceName = res.getString(resId);
+            }catch (Exception e1) {
```
Comment (jameswwarner) [inline survata-sdk/src/main/java/com/survata/utils/Utils.java]: } catch(...) with space

**Comment by jameswwarner** [comment]
Works for me. 🚢

**Comment by jameswwarner** [comment]
🔨 & 🚢 from me

**Pull request merged**